### PR TITLE
Add support for flavor replication.

### LINKED ISSUE #27: Add support for recreating Nova flavors.
Add support for recreating Nova flavors from source to dest env in a dedicated `coriolis-openstack-util replicate flavors` subcommand.
Additionally, add this functionality as an optional flag in the `migrate tenant` and `migrate batch` subcommands as well.

### PR DESCRIPTION
**Pull request by atoaca**
Resolves #27

### CODE REVIEW

File: coriolis_openstack_utils/actions/flavor_actions.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    NEW_FLAVOR_DESCRIPTION = (
+        "Created by the Coriolis OpenStack utilities for source flavor"
+        " '%s'.")
+
+    @property
+    def flavor_name_format(self):
+        return CONF.destination.new_flavor_name_format
+
+    def check_same_flavor(self, src_flavor, dest_flavor):
+        relevant_keys = ['ram', 'disk', 'vcpus', 'rxtx_flavor']
```
Comment (aznashwan) [inline coriolis_openstack_utils/actions/flavor_actions.py]: s/rxtx_flavor/rxtx_factor/g
add `swap`

File: coriolis_openstack_utils/actions/flavor_actions.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+    def create_flavor_body(self):
+        src_flavor = self._source_openstack_client.nova.flavors.get(
+            self.payload['src_flavor_id'])
+        relevant_keys = ['ram', 'disk', 'vcpus', 'swap', 'rxtx_factor']
+        src_flavor_info = src_flavor.to_dict()
+        body = {k: v for k, v in src_flavor_info.items()
+                if k in relevant_keys}
+        nova_client = self._destination_openstack_client.nova
+        if nova_client.api_version.get_string() >= '2.55':
```
Comment (aznashwan) [inline coriolis_openstack_utils/actions/flavor_actions.py]: a) double-check that the Nova client automatically uses the latest microversion (and doesn't default to '2.0' or whatever
b) either cast both to floats or use `oslo_version` (not `oslo_versionedobject`) and then do the comparison (do not compare version strings unless they're timestamps or something with a fixed structure!)

File: coriolis_openstack_utils/actions/flavor_actions.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        for tenant_name in dest_tenant_names:
+            try:
+                tenant_id = self._destination_openstack_client.get_project_id(
+                        tenant_name)
+                flavor_access.add_tenant_access(dest_flavor_id, tenant_id)
+                LOG.info("Adding access for tenant \"%s\" to flavor \"%s\""
+                         % (tenant_id, dest_flavor_id))
+            except keystoneauth1.exceptions.http.Forbidden:
+                LOG.warn("Cannot fetch id of destination tenant named \"%s\" "
+                         "due to lack of priviledges of current user."
```
Comment (aznashwan) [inline coriolis_openstack_utils/actions/flavor_actions.py]: Space between '.' and EOL '"'

File: coriolis_openstack_utils/actions/flavor_actions.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        dest_flavor = self._destination_openstack_client.nova.flavors.create(
+            **body)
+        if not dest_flavor.is_public:
+            self.add_tenant_access_to_flavor(dest_flavor)
+
+        return dest_flavor.to_dict()
+
+    def cleanup(self):
+        dest_flavor = self._destination_openstack_client.nova.flavors.find(
+            name=self.get_new_flavor_name(), is_public=None)
```
Comment (aznashwan) [inline coriolis_openstack_utils/actions/flavor_actions.py:178]: What's the reasoning behind explicitly setting `is_public=None`? (won't it have a similar semantic to `is_public=False` considering that bool(None) == False?)
(Do please test what happens with `is_public=False` before updating the PR)

File: coriolis_openstack_utils/actions/tenant_actions.py [hunk 5]
```diff
@@ -334,6 +335,18 @@ def execute_operations(self):
...
+            src_flavors = [flavor.id for flavor in
+                           self._source_openstack_client.nova.flavors.list(
+                               is_public=None)]
+            for src_flavor_id in src_flavors:
+                dest_client = self._destination_openstack_client
+                flavor_migration_action = flavor_actions.FlavorCreationAction(
+                    {'src_flavor_id': src_flavor_id},
+                    source_openstack_client=self._source_openstack_client,
+                    destination_openstack_client=dest_client,
+                     )
```
Comment (aznashwan) [inline coriolis_openstack_utils/actions/tenant_actions.py]: Please no trailing parens (and either way it's over-indented), please put it in the same line as the last kwarg.

**Review by aznashwan — CHANGES_REQUESTED** [review]
LGTM apart from some minor observations.

File: coriolis_openstack_utils/actions/flavor_actions.py (hunk 4 above)
Comment (atoaca) [reply coriolis_openstack_utils/actions/flavor_actions.py:178]: is_public=None accounts for both is_public=False(private flavors) and is_public=True(public flavors)

Comment (aznashwan) [reply coriolis_openstack_utils/actions/flavor_actions.py:178]: Makes sense, gg.

**Review by aznashwan — APPROVED** [review]

**Pull request merged**